### ISSUE TITLE
Same data give different loss.

### ISSUE THREAD

**Issue by Mudryi** [catboost/catboost#2126]
I tried to overfit the model for specific data, so I use it as a train and validation data same Pool, but during the training, all the losses have different values, but the data is absolutely the same.
Here is code: 
```
train_dataset = Pool(data=X_train,
                     label=y_train,
                     cat_features=categorical_features_indices)

CatBoost_model = CatBoostClassifier(iterations=3000,
                                    eval_metric='BrierScore',
                                    custom_metric=['AUC', 'PRAUC', 'Recall'],
                                    loss_function='Logloss',
                                    boosting_type='Plain',
                                    depth=6,
                                    learning_rate=0.4,
                                    scale_pos_weight=scale_weight,
                                    early_stopping_rounds=50,
                                    random_seed=1234
                                    )

CatBoost_model.fit(train_dataset,
                   eval_set=train_dataset,
                   use_best_model=True,
                   verbose=False,
                   plot=True
                   )
```
Here is a screenshot of the training process:
![Screenshot 2022-07-08 at 20 57 36](https://user-images.githubusercontent.com/42421131/178045473-3e241367-c322-4c7a-a16e-a7e31104554c.png)

catboost version: test on both '1.0.4' and '1.0.6'
Operating System: macOS Monteray

**Comment by Mudryi** [comment]
Thanks, now I got it.
